static/src/javascripts/projects/commercial/modules/liveblog-adverts.ts

```diff
@@ -167,11 +168,23 @@ export const init = (): Promise<void> => {
 
 	return fastdom
 		.measure(() => {
+			const isServerSideAdsMode =
+				document.querySelector('.ad-slot--inline1') !== null;
+			if (isServerSideAdsMode) {
+				throw Error;
```

Comment:
Control flow with errors might not be ideal. At the very least, you would want to create a custom Error and making sure that in your catch block the error is an instance of that class.

https://javascript.info/custom-errors

Reply:
Thanks Max. I've moved the check outside of the promise chain.